modules/auxiliary/scanner/mssql/mssql_schemadump.rb

```diff
@@ -29,9 +31,12 @@ def initialize
   end
 
   def run_host(ip)
+    if session
+      set_session(session.client)
+    end
```

Comment:
Can't comment across all the line here due to GitHub reasons 😄 

Could these be combined maybe?
```suggestion
    if session
      set_session(session.client)
    else
      unless mssql_login_datastore
      print_error("#{datastore['RHOST']}:#{datastore['RPORT']} - Invalid SQL Server credentials")
      return
    end
```

When we already check the session above, we could handle the code below as part of the `else`. For me it just reads better. Not super strongly opinionated though 😄